### DISCUSSION TITLE
error module parse failed webpack-loader.js

### DISCUSSION THREAD

**Discussion by DanitRoss** [cypress-io/cypress#25322] [Questions and Help]
Hi !
i'm having issue trying to run my first component test. 
This is the full error: 
```

The following error originated from your test code, not from Cypress.

> Module parse failed: Unexpected token (30:8)
File was processed with these loaders:
* ./node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js
You may need an additional loader to handle the result of these loaders.
| // Alternatively, can be defined in cypress/support/component.d.ts
| // with a <reference path="./component" /> at the top of your spec.
> declare global {
|     namespace Cypress {
|         interface Chainable {

When Cypress detects uncaught errors originating from your test code it will automatically fail the current test.

Cypress could not associate this error to any specific test.

We dynamically generated a new test to display this failure
```

This is my test:

`it('uses custom text for the button label', () => {
        expect(true).to.be.true;
});`

Thanks! (:

**Comment by portyu9** [comment]
Hi @DanitRoss 

This error means Webpack is trying to parse a **TypeScript declaration as JavaScript**:

```text
Module parse failed: Unexpected token
> declare global {
```

`declare global`, `namespace Cypress`, and `interface Chainable` are TypeScript syntax. They must not be placed in a `.js` file that Angular's Webpack/Babel pipeline treats as normal JavaScript.

Move the declaration into a `.d.ts` file, for example:

```ts
// cypress.d.ts
import { mount } from 'cypress/angular'

declare global {
  namespace Cypress {
    interface Chainable {
      mount: typeof mount
    }
  }
}

export {}
```

Then make sure the file is included by the relevant `tsconfig.json`:

```json
{
  "include": [
    "src",
    "cypress/**/*.ts",
    "cypress.d.ts"
  ]
}
```

Keep the actual command registration in your runtime support file, for example:

```ts
// cypress/support/component.ts
import { mount } from 'cypress/angular'

Cypress.Commands.add('mount', mount)
```

If the declaration file has **no imports or exports**, it can instead be an ambient declaration and use:

```ts
declare namespace Cypress {
  interface Chainable {
    // custom commands
  }
}
```

The important part is that the type declaration must be processed by TypeScript rather than sent through the JavaScript parser as ordinary runtime code.

Please mark this answer as accepted if it helped, thank you.
